src/styles/_catalog.scss

```diff
@@ -1,22 +1,45 @@
+$card-height: 300px;
+$card-width: 200px;
+$gap-size: 48px;
+$catalog-max-width: $card-width * 4 + $gap-size * 3;
+$text-hover-color: #34568b;
+
 .catalog {
-  display: flex;
-  flex-wrap: wrap;
+  display: grid;
+  justify-content: center;
+  align-content: center;
+  justify-items: center;
+  align-items: center;
 
-  max-width: 944px;
-  margin: 0 auto;
+  grid-template-columns: repeat(4, $card-width);
+  grid-auto-rows: $card-height;
+  gap: $gap-size;
 
   &__card {
     display: flex;
     justify-content: center;
     align-items: center;
-    width: 200px;
-    height: 300px;
+    width: $card-width;
+    height: $card-height;
     border: 1px solid #ccc;
     border-radius: 5px;
-    margin-bottom: 48px;
+
+    transform-origin: center;
+    transition: color 0.3s, transform 0.3s;
+
+    &:hover {
+      color: $text-hover-color;
+      transform: scale(1.2);
+    }
   }
 
-  &__card:not(:nth-child(4n)) {
-    margin-right: 48px;
+  @media (max-width: 487px) {
+    grid-template-columns: 1fr;
+  }
+  @media (min-width: 488px) and (max-width: 767px) {
+    grid-template-columns: 1fr 1fr;
+  }
+  @media (min-width: 768px) and (max-width: 1023px) {
+    grid-template-columns: 1fr 1fr 1fr;
```

Comment:
You can use only 1 media query with one constraint

Reply:
Hint: try to use auto-fill

Reply:
@MaksimusCRS hmm, interesting)) i could try XD)

Reply:
@MaksimusCRS so did you mean this:
```
  @media (max-width: 1023px) {
    grid-template-columns: repeat(auto-fill, $card-width);
  }
```
?))

Reply:
![image](https://user-images.githubusercontent.com/89999954/185887150-f2da00ba-3b4a-4775-bc6c-a8ba748ccbf0.png)
in this case i have only one `@media` query but i have other problems, as on the picture)

Reply:
other comments i fixed =)

Reply:
Ok, so you can add one more media query and it would be better then your previous solution :slightly_smiling_face:

Reply:
Actually you don't need 2 constraints anywhere, it's main mistake here

Reply:
@MaksimusCRS as i reallized and @OleksiiBlezniuk said on QnA - using min-max we creating a range of spesified breakpoints for different kinds of divice, same if i dont use the exact min-max for other diveces it wont work correctly, like it will change my grid because of content size but not of screen size) So I choosed this way just because its sounds right and the result will be more expected, am i wrong?)

Reply:
@MaksimusCRS yeah, I agree i could use `@media (max-width: 767px)` without `and  (min-width: 488px)`  and it will work absolutley the same)